File: solutions/go/jedliks-toys/1/jedliks_toys.go
package jedlik

import (
    "fmt"
)


func (car *Car) Drive() {
	if car.battery < car.batteryDrain {
        fmt.Println("Not enough battery to drive")
        return
    }
    car.distance += car.speed
    car.battery -= car.batteryDrain
}

func (car Car) CanFinish(trackDistance int) bool {
	maxDrives := car.battery / car.batteryDrain
    maxDistance := maxDrives * car.speed
    return maxDistance >= trackDistance
}

func (car Car) DisplayDistance() string {
    distance := fmt.Sprintf("Driven %v meters", car.distance)
    return distance
}

func (car Car) DisplayBattery() string {
    battery := fmt.Sprintf("Battery at %v%%", car.battery)
    return battery
}

File: solutions/csharp/jedliks-toys/1/JedliksToys.cs
class RemoteControlCar
{
    public int Distance {get; private set; } = 0;
    public int Battery { get; private set; } = 100;
    
    public static RemoteControlCar Buy()
    {
        return new RemoteControlCar();
    }

    public string DistanceDisplay()
    {
        string distance = $"Driven {Distance} meters";
        return distance;
    }

    public string BatteryDisplay()
    {
        if (Battery <= 0)
        {
            return "Battery empty";
        }
        else
        {
            return $"Battery at {Battery}%";
        }
        
    }

    public void Drive()
    {
        if (Battery > 0)
        {
            Distance += 20;
            Battery -= 1;
        }
    }
}
